### PR TITLE
fix(drop-down-toggle): migrate to be ref forwarder

### PR DESCRIPTION
**Pull request by hetpatel33**
Relevant to #4194 
@bpas247 @taion @mxschmitt

### CODE REVIEW

File: src/DropdownToggle.js [hunk 1]
```diff
@@ -7,80 +7,82 @@ import BaseDropdownToggle from 'react-overlays/DropdownToggle';
...
-    } = this.props;
+const defaultProps = {
+  as: Button,
+};
 
-    if (childBsPrefix !== undefined) {
-      props.bsPrefix = childBsPrefix;
-    }
+const DropdownToggle = React.forwardRef((props, ref) => {
+  const {
```
Comment (taion) [inline src/DropdownToggle.js]: can you spread this out in the function signature, and name the rest param `props` instead of `otherProps`?

Comment (hetpatel33) [reply src/DropdownToggle.js]: Sure @taion 
Can you also help me with adding the ref to the`BaseDropdownToggle` which is a functional component?
Should I pass it directly to the children?

**Comment by jquense** [comment]
heads up this is already handled in https://github.com/react-bootstrap/react-bootstrap/pull/4661

**Comment by hetpatel33** [comment]
@jquense @taion  Why no one is updating #4194?
Can you help me finding the issues which are up for pick?

File: src/DropdownToggle.js (hunk 1 above)
Comment (taion) [reply src/DropdownToggle.js]: i think we need #4661, then, sorry

**Comment by mxschmitt** [comment]
Closed because it will be already implemented in #4661. But @hetpatel33 thanks for your work anyway.

**Pull request closed without merging**